_posts/2019-09-11-how-to-stay-up-to-date.markdown

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: How to Stay Up to Date with Your Rails Application
+date: 2019-08-27 13:56:00
+categories: ["Rails", "Upgrades"]
+author: bronzdoc
+---
+
+An outdated Rails application doesn't happen overnight. In FastRuby.io we work with a lot of clients who have outdated Rails applications and we help them upgrade to a newer Rails version. In this article I share some things that you could start doing to avoid falling out of date.
+
+<!--more-->
+
+### Stop monkey patching
+
+In our experience having monkey-patched gems is usually one of the hardest things to deal with. We have to spend hours updating monkey-patched gems to make them compatible with newer Rails APIs. So please keep that in mind before monkey patching Rails core libraries or gems that depend on specific Rails versions.
+
+### Start treating deprecations warnings as errors
+
+In Rails you can configure this option, so everytime a deprecation warning occurs you can address it as soon as you can.
+This will help you to fix the issue immediately and be prepared for the next version jump.
+
+You will need to add the following configuration in your `config/environments/test.rb` and `config/environments/test.rb` files.
+
+```ruby
+  # Raise error on deprecation¬
+  config.active_support.deprecation = :raise
+```
+
+### Dual Boot
+
+One of the first things we do at FastRuby.io when we start a new Rails upgrade project is to setup a "Dual Boot" Gemfile.
+What this setup will do is allow you to have two Gemfile.lock files, one for the latest version of Rails and one for your current version of Rails.
+
+With this setup you will be able to configure your CI to run both versions of Rails. As long as you "mark" the Rails master build job as "optional" it won't block your PR merges. You will have the chance to see how your test behaves with the latest version of Rails.
+
+We even wrote a [blog post](https://www.fastruby.io/blog/upgrade-rails/dual-boot/dual-boot-with-rails-6-0-beta.html) about this setup, you should check it out.
+
+### Check for outdated gems
+
+You can use bundle outdated to check outdated gems in your current bundle. While `bundle outdated` is quite useful it doesn't give us details relative to our own `Gemfile`. In Ombulabs we like to use  [next_rails](https://rubygems.org/gems/next_rails/versions/1.0.0) because this gem will give us a set of tools to check for outdated gems, check incompatibilities and even bootstrap an upgrade project.
```

Comment:
```suggestion
You can use bundle outdated to check outdated gems in your current bundle. While `bundle outdated` is quite useful it doesn't give us details relative to our own `Gemfile`. In Ombu Labs we like to use  [next_rails](https://rubygems.org/gems/next_rails/versions/1.0.0) because this gem will give us a set of tools to check for outdated gems, check incompatibilities and even bootstrap an upgrade project.
```